configparser/parse_test.go

```diff
@@ -0,0 +1,109 @@
+package configparser_test
+
+import (
+	"net/url"
+	"path"
+	"reflect"
+	"testing"
+	"time"
+
+	"github.com/benchttp/runner/config"
+	"github.com/benchttp/runner/configparser"
+)
+
+const (
+	testdataConfigPath = "../test/testdata/config"
+	testURL            = "http://localhost:9999?fib=30&delay=200ms"
+)
+
+var supportedExt = []string{
+	".yml",
+	".yaml",
+	".json",
+}
+
+// TestParse ensures the config file is open, read, and correctly parsed.
+func TestParse(t *testing.T) {
+	t.Run("happy path for all extensions", func(t *testing.T) {
+		for _, ext := range supportedExt {
+			expCfg := newExpConfig()
+			fname := path.Join(testdataConfigPath, "benchttp"+ext)
+
+			gotCfg, err := configparser.Parse(fname)
+			if err != nil {
+				// critical error, stop the test
+				t.Fatal(err)
+			}
+
+			gotURL, expURL := gotCfg.Request.URL, expCfg.Request.URL
+
+			// compare *url.URLs separately, as they contain unpredictable values
+			// they need special checks
+			if !sameURL(gotCfg.Request.URL, expCfg.Request.URL) {
+				t.Errorf(
+					"unexpected parsed URL: exp %v, got %v",
+					expCfg.Request.URL, gotCfg.Request.URL,
+				)
+			}
```

Comment:
Wouldn't it be simpler to use `gotURL `and `expURL `here instead of `gotCfg.Request.URL` and `expCfg.Request.URL`?